### PR TITLE
Sweeping update of gradle and dependency versions.

### PR DESCRIPTION
**Pull request by dbrant**
This updates all of the dependencies of the project to their latest versions. Everyone should please upgrade to the latest version of Android Studio, update your build tools, and keep your version of Gradle and the Kotlin plugin up to date.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,38 +5,22 @@ addons:
...
   licenses:
     - 'android-sdk-license-.+'
 
-before_script:
-  - echo no | android create avd --force -n test -t $ANDROID_TARGET --abi $ANDROID_ABI
-  - emulator -avd test -no-audio -no-window -no-boot-anim &
-  - android-wait-for-emulator
-
 script:
-  - ./gradlew clean check connectedCheck jacocoTestReport
```
Comment (whym) [inline .travis.yml]: I believe this change removes connected tests and code coverage checks (tests under app/src/androitTests) from Travis build scripts. Why do they have to go?

**Comment by misaochan** [comment]
Note to the person testing this: Please make sure that you give the app a thorough overall test on the ProdRelease build variant, as we have had issues with library and gradle updates causing issues in Release builds that wasn't apparent on Debug builds.

File: .travis.yml (hunk 1 above)
Comment (dbrant) [reply .travis.yml]: @whym  whoops, I've put this back in the subsequent commits. I think I must have removed this while getting stuff working locally.

**Comment by whym** [comment]
This has now conflicts to be resolved. @dbrant per 2884bd9 I assume we want to get rid of multidex, right?

**Comment by dbrant** [comment]
It looks like something is pushing the app beyond the dex limit again...

**Comment by dbrant** [comment]
Ready! Note to reviewers: this will now introduce Proguard (for minifying only, not obfuscating) for all builds, including debug, release, and test. This should not have any adverse effects, but there might be an infinitesimal possibility of compile-time or runtime errors if something goes wrong in one or more dependencies.

Note that this also greatly reduces your method count, which will now be a healthy ~45000, allowing you to add dozens more custom libraries if you like 😉.

**Comment by misaochan** [comment]
Thanks for fixing the conflicts, @dbrant ! 

AFAIK, we disabled Proguard minify due to this issue: https://stackoverflow.com/questions/40232404/google-play-apk-and-android-studio-apk-usb-debug-behaving-differently . Basically, location-based category suggestions were not working when we had minify enabled. However, that code has been refactored significantly since then, so I'm all good with having minify back, as long as location-based category suggestions are tested successfully on a Release build variant. :) 

@neslihanturan do you think you could do this? Build a prodRelease build, upload a geotagged image for a well-known location, and see if appropriate category suggestions are given?

**Comment by neslihanturan** [comment]
@dbrant thank you! Category suggestion works on prodRelease, so I merge this PR.

**Pull request merged**